Source/Relativity.Test.Helpers/Relativity.Test.Helpers/UserHelpers/CreateUser.cs

```diff
@@ -14,235 +14,282 @@ public static class CreateUser
+				artifactId = EnsureSuccess<Choice>(choiceQueryResult);
+
+				return artifactId;
 			}
-			else
+			catch (Exception ex)
 			{
-				throw new TestHelpersException($"Error Finding Choice Artifact Id [{nameof(choiceType)}:{choiceType},{nameof(value)}:{value}]");
+				throw new TestHelpersException("Error Finding Choice Artifact", ex);
 			}
-
-			return artifactId;
 		}
 
 		public static int FindGroupArtifactId(IRSAPIClient proxy, string group)
 		{
-			int artifactId = 0;
-
-			Query<kCura.Relativity.Client.DTOs.Group> queryGroup =
-					 new Query<kCura.Relativity.Client.DTOs.Group>
-					 {
-						 Condition = new TextCondition(GroupFieldNames.Name, TextConditionEnum.EqualTo, @group)
-					 };
+			try
+			{
+				int artifactId = 0;
 
-			queryGroup.Fields.Add(new FieldValue(ArtifactQueryFieldNames.ArtifactID));
+				Query<kCura.Relativity.Client.DTOs.Group> queryGroup =
+					new Query<kCura.Relativity.Client.DTOs.Group>
+					{
+						Condition = new TextCondition(GroupFieldNames.Name, TextConditionEnum.EqualTo, @group)
+					};
+				queryGroup.Fields.Add(new FieldValue(ArtifactQueryFieldNames.ArtifactID));
+				QueryResultSet<kCura.Relativity.Client.DTOs.Group> resultSetGroup = proxy.Repositories.Group.Query(queryGroup, 0);
 
-			QueryResultSet<kCura.Relativity.Client.DTOs.Group> resultSetGroup = proxy.Repositories.Group.Query(queryGroup, 0);
+				// Check for success.
+				artifactId = EnsureSuccess<kCura.Relativity.Client.DTOs.Group>(resultSetGroup);
 
-			if (resultSetGroup.Success && resultSetGroup.Results.Count == 1)
-			{
-				artifactId = resultSetGroup.Results.FirstOrDefault().Artifact.ArtifactID;
+				return artifactId;
 			}
-			else
+			catch (Exception ex)
 			{
-				throw new TestHelpersException($"Error Finding Group Artifact Id [{nameof(group)}:{group}]");
+				throw new TestHelpersException("Error Finding Group Artifact", ex);
 			}
-
-			return artifactId;
 		}
 
 		public static int FindClientArtifactId(IRSAPIClient proxy, string group)
 		{
-			int artifactId = 0;
+			try
+			{
+				int artifactId = 0;
+
+				Query<Client> queryClient = new Query<Client>
+				{
+					Condition = new TextCondition(ClientFieldNames.Name, TextConditionEnum.EqualTo, @group),
+					Fields = FieldValue.AllFields
+				};
+				QueryResultSet<Client> resultSetClient = proxy.Repositories.Client.Query(queryClient, 0);
+
+				// Check for success.
+				artifactId = EnsureSuccess<Client>(resultSetClient);
 
-			Query<Client> queryClient = new Query<Client>
+				return artifactId;
+			}
+			catch (Exception ex)
 			{
-				Condition = new TextCondition(ClientFieldNames.Name, TextConditionEnum.EqualTo, @group),
-				Fields = FieldValue.AllFields
-			};
+				throw new TestHelpersException("Error Finding Client Artifact", ex);
+			}
+		}
 
-			QueryResultSet<Client> resultSetClient = proxy.Repositories.Client.Query(queryClient, 0);
+		private static int EnsureSuccess<T>(this ResultSet<T> result) where T : kCura.Relativity.Client.DTOs.Artifact

```

Comment:
this should probably be moved to an ExtensionMethods class so that it can be used commonly by all RASPI calls.